### PR TITLE
Migrate to the new SDKs

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
Migrate to the new SDKs
Upgrade tooling
Add missing features

Closes https://github.com/PostHog/posthog-flutter/issues/23
Closes https://github.com/PostHog/posthog-flutter/issues/38
Closes https://github.com/PostHog/posthog-flutter/issues/46

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [ ] No breaking change or entry added to the changelog.

### CODE REVIEW

File: example/ios/Runner/Info.plist [hunk 1]
```diff
@@ -42,11 +44,9 @@
 	<key>com.posthog.posthog.POSTHOG_HOST</key>
 	<string>https://app.posthog.com</string>
 	<key>com.posthog.posthog.API_KEY</key>
-	<string>YOUR_API_KEY_GOES_HERE</string>
+	<string>_6SG-F7I1vCuZ-HdJL3VZQqjBlaSb1_20hDPwqMNnGI</string>
```
Comment (pauldambra) [inline example/ios/Runner/Info.plist:47]: should this be hard-coded?

File: example/pubspec.yaml [hunk 2]
```diff
@@ -1,44 +1,65 @@
...
+    #   posthog_flutter: ^x.y.z
+    # See https://dart.dev/tools/pub/dependencies#version-constraints
+    # The example app is bundled with the plugin so we use a path dependency on
+    # the parent directory to use the current plugin's version.
     path: ../
 
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
-  cupertino_icons: ^1.0.3
+  cupertino_icons: ^1.0.2
```
Comment (pauldambra) [inline example/pubspec.yaml:31]: does it matter this is a downgrade?

File: example/web/index.html [hunk 3]
```diff
@@ -1,14 +1,67 @@
...
+    !function(t,e){var o,n,p,r;e.__SV||(window.posthog=e,e._i=[],e.init=function(i,s,a){function g(t,e){var o=e.split(".");2==o.length&&(t=t[o[0]],e=o[1]),t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}}(p=t.createElement("script")).type="text/javascript",p.async=!0,p.src=s.api_host+"/static/array.js",(r=t.getElementsByTagName("script")[0]).parentNode.insertBefore(p,r);var u=e;for(void 0!==a?u=e[a]=[]:a="posthog",u.people=u.people||[],u.toString=function(t){var e="posthog";return"posthog"!==a&&(e+="."+a),t||(e+=" (stub)"),e},u.people.toString=function(){return u.toString(1)+".people (stub)"},o="capture identify alias people.set people.set_once set_config register register_once unregister opt_out_capturing has_opted_out_capturing opt_in_capturing reset isFeatureEnabled onFeatureFlags getFeatureFlag getFeatureFlagPayload reloadFeatureFlags group updateEarlyAccessFeatureEnrollment getEarlyAccessFeatures getActiveMatchingSurveys getSurveys".split(" "),n=0;n<o.length;n++)g(u,o[n]);e._i.push([i,s,a])},e.__SV=1)}(document,window.posthog||[]);
+    posthog.init('phc_pQ70jJhZKHRvDIL5ruOErnPy6xiAiWCqlL4ayELj4X8', {api_host: 'https://app.posthog.com'})
```
Comment (pauldambra) [inline example/web/index.html:44]: same here does it matter that the key is now hard coded?

**Review by pauldambra — APPROVED** [review]
almost all beyond me 🤣 I trust you've tested it 🥇 

does this mean Android flutter apps would be able to test screen recording?

File: example/ios/Runner/Info.plist (hunk 1 above)
Comment (marandaneto) [reply example/ios/Runner/Info.plist:47]: It's just our example project within the repo, so you can run and test stuff right away, its one of our testing projects key.

File: example/pubspec.yaml (hunk 2 above)
Comment (marandaneto) [reply example/pubspec.yaml:31]: Nope, `^` gets the latest version anyway, the tooling did it automatically.

Comment (marandaneto) [reply example/pubspec.yaml]: I reverted just because it was done by the tooling anyway

**Comment by marandaneto** [comment]
> almost all beyond me 🤣 I trust you've tested it 🥇
> 
> does this mean Android flutter apps would be able to test screen recording?

Nope, Flutter is its own thing, recording has to be done from scratch.

**Pull request merged**